### DISCUSSION TITLE
Welcome to Discussions!

### DISCUSSION THREAD

**Discussion by mr-manuel** [mr-manuel/venus-os_dbus-enphase-envoy#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by RudeManagement** [comment]
Hi Manuel,

Thanks for this script! I've spend the night trying to make it work, but no luck.
My configuration is a Cerbo GX on http://192.168.1.49/ root enabled and MQTT (if first tried only SSL, and also plaintext) and an Envoy on http://192.168.1.50.
I retrieved a very long installer password for the Envoy and modify the config.ini with it.
I changed the data (11 IQ7, etc), enable MQTT publishing "enable = 1"?), left the TLS settings alone.
Finaly I added my username for the CerboGX root connection and the password.

And then I tried all combinaison of the settings I could think of.
Using the tail command, it seems I've got a problem with the SSL connection to my envoy.

Being a noob, could you first confirm I'm putting the right credential into the right places please?
First is the looonnng pasword for the envoy and at the end the root and password for the Cerbo or is it my installer account on enphase?
 I'll go back to the initial config.ini and post the errors I get if you feel like helping :D
 
 Kind regards,

**Comment by mr-manuel** [comment]
Hi, can you login at http://192.168.1.50/installer/setup/home with username `installer` and the calculated password from https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/blob/master/tools/passwordCalc.py? The generated password should be 8 alphanumeric charaters long.

You can also insert the script [here](https://www.programiz.com/python-programming/online-compiler/) and replace the serial number `122112123456` in line 7 with your serial number.

**Reply by RudeManagement** [reply]
Thanks for your quick reply,
Yes I can access http://192.168.1.50/installer/setup/home no problem. But it didn't request any password. I created a token here: https://entrez.enphaseenergy.com/entrez_tokens

Thanks for the emulator, I now have a 8 character long password, so easy!
That's the first password of the script, right? 
What is username and password for at the end of the script? same?

Here is the debug file, but I'm not sure my config.ini is right. I attached it in txt (not ini, and hid my email and password)

[debug.txt](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/files/10474392/debug.txt)
[config.txt](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/files/10474398/config.txt)

**Reply by mr-manuel** [reply]
Yes, this is the password under `[ENVOY]` and then `password`. The `config.ini` file has different sections. As you can see the `username` and `password` at the end is under the `[MQTT]` section. So they are for the MQTT connection to the broker.

As you decribed you probably enabled MQTT via the Venus OS GUI so no username and password are needed. Therefore comment them out with a semicolon `;` at the beginning. Obviously in the `[MQTT]` section of the `config.ini`.

Which software version do you have on your Envoy? You can find it under http://192.168.1.50/home.

If you open http://192.168.1.50/production.json?details=1 in a browser, do you get redirected to https or have you modified the script?

**Reply by RudeManagement** [reply]
Okay, MQTT username and password commented.

Envoy is running D7.0.88 (5580b1)

I think I'm redircted to https, but https is in red and striked trhu. 
If I copy paste the link: https://192.168.1.50/production.json?details=1
![image](https://user-images.githubusercontent.com/121695004/213919564-36f1efeb-3098-44a7-b057-21dfe9fc6c20.png)

Thanks again for your quick and helpfull reply! Much appreciated!

**Reply by RudeManagement** [reply]
Okay, I understand why you asked about the redirection.
So I exportd the certificate from the browsr to /data/etc/dbus-enphase-envoy/envoy-12223300xxx0.crt 
And enabled tls in config.ini:
; Enables TLS
tls_enabled = 1

; Absolute path to the Certificate Authority certificate file that is to be treated as trusted by this client
tls_path_to_ca = /data/etc/dbus-enphase-envoy/envoy-122233002480.crt

; Disables verification of the server hostname in the server certificate
;tls_insecure = 1

Now the error I'm getting is:ConnectionResetError: [Errno 104] Connection reset by peer

[debug.txt](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/files/10474604/debug.txt)

**Reply by mr-manuel** [reply]
This driver do not work with the `D7.x.x` firmware since it has no direct local API anymore. It requires a token that need to be requested every few months. If you don't need the latest firmware you can open a ticket at Enphase and request a firmware downgrade to `D5.0.62 (5e57a9)`.

**Reply by RudeManagement** [reply]
Nice, thanks a lot for your time!
I'll do that then :)

**Reply by mr-manuel** [reply]
You're welcome

**Reply by mr-manuel** [reply]
By the way. The settings you mentioned above are from the `[MQTT]` settings and are only for MQTT.

**Reply by mr-manuel** [reply]
@RudeManagement did you solve it by downgrading?

**Comment by RudeManagement** [comment]
Hi,
No, enphase refuses to do it :/

**Reply by mr-manuel** [reply]
Unfortunately I have no other Envoy with D7 firmware to make some testing, else I could figure it out. If you find some scripts that are polling the data successfully from Envoy with D7 firmware let me know.

**Comment by RudeManagement** [comment]
Sure, will do
I think someone figured it out. But i didn't get a chance to look into it

**Reply by mr-manuel** [reply]
If you send me the link to the code I can look into it

**Reply by mr-manuel** [reply]
I found now code that I can use to connect to D7 firmware. Unfortunately I have no Envoy with D7 firmware. Would it be possible that you share access to your Envoy by opening a port for my IP or via ngrok? If yes can you provide me your contact details like Discord, Telegram or E-Mail?

**Reply by RudeManagement** [reply]
Sure, I'll send you a PM when I get home :)

**Reply by mr-manuel** [reply]
I will release a version this week which supports the D5 and D7 firmware, so stay tuned :-)

**Comment by RudeManagement** [comment]
Great news!
Thanks! Sorry I didn't reply, I was fine tuning my install 🙏

**Comment by michel-pet** [comment]
thanks for the software.
it works 99% with the d7 setup.
on the old screen i don't see a pv_power_l1 screen.
strangely enough on the new screen i see the png but no data and flow.
in the txt screens this works fine.
have i set something wrong or is there something wrong in the software???.

greetings michel.
